whois-scheduler/src/main/java/net/ripe/db/whois/scheduler/task/grs/GrsDao.java

```diff
@@ -54,9 +57,15 @@ private void ensureInitialized() {
             JdbcRpslObjectOperations.sanityCheck(slaveJdbcTemplate);
             this.masterJdbcTemplate = masterJdbcTemplate;
             this.slaveJdbcTemplate = slaveJdbcTemplate;
+            this.transactionTemplate = new TransactionTemplate(new DataSourceTransactionManager(masterJdbcTemplate.getDataSource()));
```

Comment:
We should be adding all transactionManager in TransactionConfiguration.

Reply:
I tried but GRS sources are dynamic. Is there a way to inject GRS sources and provide a TransactionTemplate?

Or, add a static method to TransactionConfiguration like : 
```
public static TransactionManager transactionManager(final DataSource datasource) {
    return new DataSourceTransactionManager(datasource);
}
```